Zend/Optimizer/pass1.c

```diff
@@ -235,8 +235,7 @@ void zend_optimizer_pass1(zend_op_array *op_array, zend_optimizer_ctx *ctx)
 			        Z_TYPE(ZEND_OP1_LITERAL(opline)) == IS_STRING) {
 					/* for A::B */
 					if (op_array->scope &&
-						!strncasecmp(Z_STRVAL(ZEND_OP1_LITERAL(opline)),
-						ZSTR_VAL(op_array->scope->name), Z_STRLEN(ZEND_OP1_LITERAL(opline)) + 1)) {
+						zend_string_equals_literal(Z_STR(ZEND_OP1_LITERAL(opline)), op_array->scope->name)) {
```

Comment:
Isn't this supposed to be a case insensitive comparison? Also this is not comparing with a literal...

Reply:
Yeah... removed the wrong word